### PR TITLE
채팅 목록 조회 시 offset방식으로 변경

### PR DESCRIPTION
**Pull request by he2joojo**
## Issues
- #115 

## What is this PR? 👓
채팅 목록 조회 시 offset방식으로 변경에 대한 pr입니다.

## Key changes 🔑
- 요청 시 `page`와 `size`를  `Pageable`로 받습니다.
- 응답 시 `nextCursor`에 다음 페이지 번호를 넣었습니다.

### CODE REVIEW

File: src/main/java/kr/codesquad/secondhand/presentation/ChatController.java [hunk 1]
```diff
@@ -52,12 +55,22 @@ public DeferredResult<ApiResponse<ChatLogResponse>> readAll(
...
 
     @GetMapping("/chats")
-    public ApiResponse<CustomSlice<ChatRoomResponse>> readList(
-            @RequestParam(required = false) Long cursor,
-            @RequestParam(required = false, defaultValue = "10") int size,
-            @Auth Long memberId
-    ) {
-        return new ApiResponse<>(HttpStatus.OK.value(), chatRoomService.read(cursor, size, memberId));
+    public DeferredResult<ApiResponse<CustomSlice<ChatRoomResponse>>> readList(
+            Pageable pageable,
```
Comment (23Yong) [inline src/main/java/kr/codesquad/secondhand/presentation/ChatController.java]: `@PageableDefault` 애노테이션을 통해 기본 값들을 처리할 수 있을 것 같아요!

File: src/main/java/kr/codesquad/secondhand/repository/chat/querydsl/ChatPaginationRepository.java [hunk 2]
```diff
@@ -36,13 +35,21 @@ public Slice<ChatRoomResponse> findByMemberId(Long memberId, Long chatRoomId, in
...
                 .fetch();
-        return checkLastPage(pageSize, chatRoomResponses);
+
+        Integer hasNext = queryFactory
+                .selectOne()
+                .from(chatRoom)
+                .where(equalsMemberId(memberId),
+                        chatRoom.lastSendTime.before(
+                                chatRoomResponses.get(chatRoomResponses.size() - 1).getLastSendTime()))
+                .fetchOne();
```
Comment (23Yong) [inline src/main/java/kr/codesquad/secondhand/repository/chat/querydsl/ChatPaginationRepository.java]: 41라인 limit에서 +1 해서 한 개 더 가져온 다음에 pageSize보다 1크면 true, 아니면 false는 어떠신가용?

**Review by 23Yong — APPROVED** [review]
수고하셨습니다!

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [68.37%]|:green_apple:|
|:-|:-:|:-:|
|[ChatPaginationRepository.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/0ca89faaf3b435c0d221520c3f6b5ca55f42abe9/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Frepository%2Fchat%2Fquerydsl%2FChatPaginationRepository.java)|100%|:green_apple:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/0ca89faaf3b435c0d221520c3f6b5ca55f42abe9/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|89.6%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/0ca89faaf3b435c0d221520c3f6b5ca55f42abe9/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|29.89%|:x:|

|Total Project Coverage|80.84%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/kr/codesquad/secondhand/presentation/ChatController.java (hunk 1 above)
Comment (he2joojo) [reply src/main/java/kr/codesquad/secondhand/presentation/ChatController.java]: 우와..ㅇ예외처리 다 따로 해줘야하는줄 알고 다음에 해야지 하고 넘어갔는데 기술 미쳤네용.. 브루니짱 다른 조도 이거 쓴곳 없었는데 자랑하고 다닐게여~~

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [77.72%]|:green_apple:|
|:-|:-:|:-:|
|[ChatPaginationRepository.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a757319b76c84d3df6b503a23f52d4624b1a2977/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Frepository%2Fchat%2Fquerydsl%2FChatPaginationRepository.java)|100%|:green_apple:|
|[ItemService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a757319b76c84d3df6b503a23f52d4624b1a2977/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fitem%2FItemService.java)|92.33%|:green_apple:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a757319b76c84d3df6b503a23f52d4624b1a2977/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|89.6%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a757319b76c84d3df6b503a23f52d4624b1a2977/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|29.89%|:x:|

|Total Project Coverage|80.85%|:green_apple:|
|:-|:-:|:-:|

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [78.72%]|:green_apple:|
|:-|:-:|:-:|
|[ChatPaginationRepository.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/c832fb8237a2b97392560b818b19395c8a957e6a/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Frepository%2Fchat%2Fquerydsl%2FChatPaginationRepository.java)|100%|:green_apple:|
|[ImageService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/c832fb8237a2b97392560b818b19395c8a957e6a/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fimage%2FImageService.java)|95.45%|:green_apple:|
|[ItemService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/c832fb8237a2b97392560b818b19395c8a957e6a/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fitem%2FItemService.java)|92.33%|:green_apple:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/c832fb8237a2b97392560b818b19395c8a957e6a/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|89.6%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/c832fb8237a2b97392560b818b19395c8a957e6a/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|29.89%|:x:|

|Total Project Coverage|80.9%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**